Cabal/src/Distribution/PackageDescription/Check.hs

```diff
@@ -999,14 +1006,14 @@ pd2gpd pd = gpd
 -- present in our .cabal file.
 checkMissingDocs
   :: Monad m
-  => [Glob] -- data-files globs.
-  -> [Glob] -- extra-source-files globs.
-  -> [Glob] -- extra-doc-files globs.
-  -> [Glob] -- extra-files globs.
+  => [Either FilePath Glob] -- data-files globs.
+  -> [Either FilePath Glob] -- extra-source-files globs.
+  -> [Either FilePath Glob] -- extra-doc-files globs.
+  -> [Either FilePath Glob] -- extra-files globs.
```

Comment:
the `Either` blindness here is a bit of a smell imo, probably better for `Glob` to have a `OopsAllLiterals` constructor?